lib/hd/keyorigin.js

```diff
@@ -0,0 +1,323 @@
+
+    const slashSplit = str.split('/');
+
+    assert(
+      slashSplit[0].length === 8,
+      `Expected 8 characters fingerprint, found ${slashSplit[0].length} instead`
+    );
+
+    const fingerprintHex = slashSplit[0];
+
+    assert(
+      base16.test(fingerprintHex),
+      `Fingerprint ${fingerprintHex} is not hex`
+    );
+
+    const fingerPrint = parseInt(fingerprintHex, 16);
+    const path = common.parsePathToArray(slashSplit.slice(1), true);
+
+    return this.fromOptions({fingerPrint, path});
+  }
+
+  static fromString(str) {
+    return new this().fromString(str);
+  }
+
+  /**
+   * Test whether two KeyOriginInfo objects are equal.
+   * @param {KeyOriginInfo} keyInfo
+   * @returns {Boolean}
+   */
+
+  equals(keyInfo) {
+    if (!KeyOriginInfo.isKeyOriginInfo(keyInfo)) {
+      return false;
+    }
+
+    if (this.fingerPrint !== keyInfo.fingerPrint) {
+        return false;
+    }
+
+    if (this.path.length !== keyInfo.path.length) {
+        return false;
+    }
+
+    for (let i = 0; i < this.path.length; i++) {
+      if (this.path[i] !== keyInfo.path[i]) {
+            return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Convert KeyOriginInfo to a more user-friendly object.
+   * @returns {Object}
+   */
+
+  inspect() {
+    return this.format();
+  }
+
+  /**
+   * Convert KeyOriginInfo  to a more user-friendly object.
+   * @returns {Object}
+   */
+
+  format() {
+    let path = 'm';
+    for (const p of this.path) {
+      const hardened = p & common.HARDENED ? '\'' : '';
+      path += `/${p & 0x7fffffff}${hardened}`;
+    }
+    return {
+      fingerPrint: this.fingerPrint.toString(16).padStart(8, '0'),
+      path
+    };
+  }
+
+   /**
+   * Inject properties from serialized data.
+   * @private
+   * @param {Buffer} data
+   * @returns {KeyOriginInfo}
+   */
+
+   fromRaw(data) {
+    return this.fromReader(bio.read(data));
+  }
+
+  /**
+   * Instantiate KeyOriginInfo from serialized data.
+   * @param {Buffer} data
+   * @returns {KeyOriginInfo}
+   */
+
+  static fromRaw(data) {
+    return new this().fromRaw(data);
+  }
+
+  /**
+   * Inject properties from buffer reader.
+   * @private
+   * @param {BufferReader} br
+   */
+
+  fromReader(br) {
+    this.fingerPrint = br.readU32BE();
+    while (br.left()) {
+      this.path.push(br.readU32());
+    }
+    return this;
+  }
+
+  /**
+   * Instantiate KeyOriginInfo from buffer reader.
+   * @param {BufferReader} br
+   * @returns {KeyOriginInfo}
+   */
+
+  static fromReader(br) {
+    return new this().fromReader(br);
+  }
+
+  /**
+   * Serialize the KeyOriginInfo.
+   * @returns {Buffer}
+   */
+
+  toRaw() {
+    const size = this.getSize();
+    return this.toWriter(bio.write(size)).render();
+  }
+
+  /**
+   * Write the KeyOriginInfo to a buffer writer.
+   * @param {BufferWriter} bw
+   */
+
+  toWriter(bw) {
+    bw.writeU32BE(this.fingerPrint);
+    for (const p of this.path) {
+      bw.writeU32(p);
+    }
```

Comment:
Either use big-endian for both fingerprint and path or use little-endian.
Same changed will be required in `fromReader()`